common/src/lib.rs

```diff
@@ -429,6 +431,8 @@ pub struct Mappings {
     pub kernel_slice_start: u64,
     /// Size of the kernel slice allocation in memory.
     pub kernel_slice_len: u64,
+    /// Start address of the kernel image relocated in memory.
+    pub kernel_image_offset: u64,
```

Comment:
We should make it clear that this is a virtual address:

```suggestion
    /// Relocation offset of the kernel image in virtual memory.
    pub kernel_image_offset: VirtAddr,
```